File: SuperiorPackGroup/BLL/ReceivingListReportBLL.vb
Imports SuperiorPackGroup.SPGReportsTableAdapters

Public Class ReceivingListReportBLL

    Private m_ReceivingListReportTableAdapter As ReceivingListReportTableAdapter = Nothing

    Public ReadOnly Property Adapter() As ReceivingListReportTableAdapter
        Get
            If (m_ReceivingListReportTableAdapter Is Nothing) Then
                m_ReceivingListReportTableAdapter = New ReceivingListReportTableAdapter
                m_ReceivingListReportTableAdapter.Connection.ConnectionString = My.Settings.UserConnectionString
            End If
            Return m_ReceivingListReportTableAdapter
        End Get
    End Property

    Public Function GetReceivingListreport(ByVal ReceivingID As Integer) As SPGReports.ReceivingListReportDataTable

        Try
            Return Adapter.GetReceivingList(ReceivingID)
        Catch ex As Exception
            MessageBox.Show(String.Format("There was an error while retrieving the information to print{0}{1}", vbCrLf, ex.Message))
            Return Nothing
        End Try

    End Function

End Class

File: SuperiorPackGroup/BLL/ReceivingListReportBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using SuperiorPackGroup.SPGReportsTableAdapters;

namespace SuperiorPackGroup
{
	public class ReceivingListReportBLL
	{

		private ReceivingListReportTableAdapter m_ReceivingListReportTableAdapter = null;

		public ReceivingListReportTableAdapter Adapter
		{
			get
			{
				if (m_ReceivingListReportTableAdapter == null)
				{
					m_ReceivingListReportTableAdapter = new ReceivingListReportTableAdapter();
					m_ReceivingListReportTableAdapter.Connection.ConnectionString = Properties.Settings.Default.UserConnectionString;
				}
				return m_ReceivingListReportTableAdapter;
			}
		}

		public SPGReports.ReceivingListReportDataTable GetReceivingListreport(int ReceivingID)
		{

			try
			{
                return Adapter.GetReceivingList(ReceivingID);
			}
			catch (Exception ex)
			{
				MessageBox.Show(string.Format("There was an error while retrieving the information to print{0}{1}", Environment.NewLine, ex.Message));
				return null;
			}

		}

	}

}